cluster/swarm/cluster.go

```diff
@@ -196,10 +199,15 @@ func (c *Cluster) RemoveImage(image *cluster.Image) ([]*dockerclient.ImageDelete
 
 // Pull is exported
 func (c *Cluster) Pull(name string, callback func(what, status string)) {
-	size := len(c.engines)
-	done := make(chan bool, size)
+	var wg sync.WaitGroup
+
+	c.RLock()
```

Comment:
I think this is troublesome - Pull() could last hours, we cannot afford to `RLock` the entire Cluster.